A28-xds-traffic-splitting-and-routing.md

```diff
@@ -0,0 +1,1095 @@
+field](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L181)
+must be
+[route](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L185),
+and its
+[cluster\_specifier](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L696):
+*   Can be
+    [Cluster](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L701)
+*   Can be
+    [Weighted\_clusters](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L719)
+    *   The sum of
+        [weights](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L278)
+        must add up to the
+        [total\_weight](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L335).
+*   Can be
+    [cluster\_header](https://github.com/envoyproxy/envoy/blob/v1.13.1/api/envoy/api/v2/route/route_components.proto#L712)
+    *   Note: the route containing this action will be ignored. This feature can
+        be accomplished by header matching.
+
+Duplicate matches are allowed, but because the first match wins, the second
+will never be used.
+
+Violating any of the validation causes the response to be rejected (NACK’ed).
+The fields not covered by this section will all be ignored.
+
+### Service config generating
+
+The service config generated by the xDS resolver will pick
+`xds_routing_experimental` as the LB policy. As an optimization, when there’s
+only one route, the LB policy picked will be `cds_experimental` or
+`weighted_target_experimental` depending on the action, so we can skip the
+`xds_routing` policy that has only one child.
+
+The `xds_routing_experimental` balancer config will contain the validated
+`{matchers, action}` pairs. The actions will be in a separate list, and will
+be referred to by the routes by their names.
+
+#### Action list
+
+All the actions from the validated `{matchers, action}` pairs will be
+de-duplicated and stored in a cache. All actions in the cache will be
+assigned a unique name.
+
+When processing an RDS response, for each action:
+
+*   If the action is cluster, the child\_policy will be `cds_experimental`, with
+    a balancer config containing the cluster string.
+*   If the action is weighted\_clusters, the child\_policy will be
+    `weighted_target_experimental`. The balancer config will contain a list of
+    `cds_experimental` LB configs, and their corresponding weights.
+*   If this action is already in the cache, it will be reused, and nothing needs
+    to be done
+*   If the action is new, it will be be added to the list and assigned a name
+    *   The name should be a human readable hash of the config, to be logged for
+        debugging purposes.
+    *   For example, the names can be `cds:cluster1` and
+        `weighted:cluster1_cluster2_<index_number>`.
```

Comment:
The example names don't look like hash of a config (as mentioned on line 431)

Reply:
Updated